apis/serving/v1beta1/inferenceservice_registry.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2021 IBM Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/kserve/modelmesh-serving/apis/serving/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// +kubebuilder:object:generate=false
+type InferenceServiceRegistry struct {
+	Client client.Client
+}
+
+func BuildPredictorWithBase(isvc *InferenceService, framework string, frameworkSpec *PredictorExtensionSpec) *v1alpha1.Predictor {
+	p := &v1alpha1.Predictor{}
+
+	p.Spec = v1alpha1.PredictorSpec{
+		Model: v1alpha1.Model{
+			Type: v1alpha1.ModelType{
+				Name: framework,
+			},
+		},
+	}
+
+	if frameworkSpec != nil && frameworkSpec.RuntimeVersion != nil {
+		p.Spec.Runtime = &v1alpha1.PredictorRuntime{
+			RuntimeRef: &v1alpha1.RuntimeRef{
+				Name: *frameworkSpec.RuntimeVersion,
+			},
+		}
+	}
```

Comment:
Interesting I guess this is good for now, maybe add an explicit comment to say that the string `runtimeVersion` is being interpreted as the name of a ServingRuntime?

Reply:
Let's see the difference between KServe and ModelMesh here:
- ModelFramework/Version: this is KServe's framework list on the `InferenceService` predictor spec, in ModelMesh this is `ModelType` on predictor
- ServingRuntime/Version:  the `runtimeVersion` currently actually is meant for the ServingRuntime's image version, so I think we should add the `servingRuntime` field on the `InferenceService` predictor spec